src/file/ShareDialog.tsx

```diff
@@ -60,25 +60,21 @@ const labelStyles = mergeStyleSets({
 
 const ShareText: React.FC = () => {
     const { scene } = useScene();
-    const [data, setData] = useState<string | undefined>(undefined);
-    const [copyMessageVisible, setMessageVisibility] = useState(false);
-    const [timeout, setMessageTimeout] = useState<number>();
-
-    useEffect(() => {
-        (async () => setData(await sceneToText(scene)))();
-    }, [scene]);
+    const data = useMemo(() => sceneToText(scene), [scene]);
+    const [copyMessageVisible, setMessageVisibility] = useBoolean(false);
+    const timerRef = useRef<number>();
 
     const doCopyToClipboard = () => {
-        navigator.clipboard.writeText(data || '');
-        setMessageVisibility(true);
-        clearTimeout(timeout);
-        setMessageTimeout(setTimeout(() => setMessageVisibility(false), 2000));
+        navigator.clipboard.writeText(data);
+        setMessageVisibility.setTrue();
+        clearTimeout(timerRef.current);
+        timerRef.current = setTimeout(() => setMessageVisibility.setFalse(), 2000);
```

Comment:
Tiny simplification:
```suggestion
        timerRef.current = setTimeout(setMessageVisibility.setFalse, 2000);
```

Reply:
I wasn't 100% confident that the reference to `setMessageVisibility` would be preserved in the timeout's context, but this does work.